harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -560,18 +558,16 @@ def _compute_validation_metrics(self) -> workload.Metrics:
                 if keys is None:
                     keys = vld_metrics.keys()
                 else:
-                    check.eq(
-                        keys,
-                        vld_metrics.keys(),
-                        "Validation metric names must match across all batches of data.",
+                    if keys != vld_metrics.keys():
+                        raise ValueError(
+                            "Validation metric names must match across all batches of data.",
+                        )
+                if not isinstance(vld_metrics, dict):
+                    raise TypeError(
+                        "validation_metrics() must return a "
+                        "dictionary of string names to Tensor "
+                        "metrics.",
```

Comment:
can you put the actual type of `vld_metrics` in the message?

Reply:
Done.